src/lib.rs

```diff
@@ -315,49 +374,135 @@ const MAVLINK_IFLAG_SIGNED: u8 = 0x01;
 
 #[derive(Debug, Copy, Clone, PartialEq, Eq)]
 // Follow protocol definition: https://mavlink.io/en/guide/serialization.html#mavlink2_packet_format
-pub struct MAVLinkV2MessageRaw {
-    pub start: u8, // = MAV_STX_V2
-    pub payload_length: u8,
-    pub incompatibility_flags: u8,
-    pub compatibility_flags: u8,
-    pub sequence: u8,
-    pub system_id: u8,
-    pub component_id: u8,
-    pub message_id: u32,
-    pub payload_buffer: [u8; 255],
-    pub checksum: u16,
+pub struct MAVLinkV2MessageRaw([u8; 1 + Self::HEADER_SIZE + 255 + 2 + Self::SIGNATURE_SIZE]);
+
+impl Default for MAVLinkV2MessageRaw {
+    fn default() -> Self {
+        Self::new()
+    }
 }
 
 impl MAVLinkV2MessageRaw {
+    const HEADER_SIZE: usize = 9;
+    const SIGNATURE_SIZE: usize = 13;
+
+    pub const fn new() -> Self {
+        Self([0; 1 + Self::HEADER_SIZE + 255 + 2 + Self::SIGNATURE_SIZE])
+    }
+
+    #[inline]
+    pub fn header(&mut self) -> &[u8] {
+        &self.0[1..=Self::HEADER_SIZE]
+    }
+
+    #[inline]
+    fn mut_header(&mut self) -> &mut [u8] {
+        &mut self.0[1..=Self::HEADER_SIZE]
+    }
+
+    #[inline]
+    pub fn payload_length(&self) -> u8 {
+        self.0[1]
+    }
+
+    #[inline]
+    pub fn incompatibility_flags(&self) -> u8 {
+        self.0[2]
+    }
+
+    #[inline]
+    pub fn compatibility_flags(&self) -> u8 {
+        self.0[3]
+    }
+
+    #[inline]
+    pub fn sequence(&self) -> u8 {
+        self.0[4]
+    }
+
+    #[inline]
+    pub fn system_id(&self) -> u8 {
+        self.0[5]
+    }
+
+    #[inline]
+    pub fn component_id(&self) -> u8 {
+        self.0[6]
+    }
+
+    #[inline]
+    pub fn message_id(&self) -> u32 {
+        u32::from_le_bytes([self.0[7], self.0[8], self.0[9], 0])
+    }
+
+    #[inline]
     pub fn payload(&self) -> &[u8] {
-        &self.payload_buffer[..self.payload_length as usize]
+        let payload_length = self.payload_length() as usize;
```

Comment:
lossless conversions-

```suggestion
        let payload_length: u8 = self.payload_length().into();
```

Reply:
The `payload_length` method return `u8`. Casting from u8 to usize is [loseless](https://doc.rust-lang.org/reference/expressions/operator-expr.html#type-cast-expressions).
> Casting from a smaller integer to a larger integer (e.g. u8 -> u32) will
>   zero-extend if the source is unsigned

I could change `payload_length` return type to `usize` if needed.

Reply:
> The `payload_length` method return `u8`. Casting from u8 to usize is [loseless](https://doc.rust-lang.org/reference/expressions/operator-expr.html#type-cast-expressions).
> > Casting from a smaller integer to a larger integer (e.g. u8 -> u32) will
> >   zero-extend if the source is unsigned
> 
> I could change `payload_length` return type to `usize` if needed.

I understand the cast is lossless. That's why you might as well use the *explicitly* lossless method.

Doing the conversion once, inside the `payload_length` method, sounds like a good approach

Reply:
the mavlink protocol defines the payload_length as `u8`, we should stay with it to make the protocol definition clear.